ios/FluentUI.Demo/FluentUIDemoTests/HUDTest_SwiftUI.swift

```diff
@@ -8,13 +8,68 @@ import XCTest
 class HUDTestSwiftUI: BaseTest {
     override var controlName: String { "HUD" }
 
+    let sleepSeconds: UInt32 = 5
+
     override func setUpWithError() throws {
         try super.setUpWithError()
         app.staticTexts["SwiftUI Demo"].tap()
+        app.buttons["Present HUD for 3 seconds"].tap()
+        // sleeps to wait for presented HUD to disappear
+        sleep(sleepSeconds)
     }
 
     // launch test that ensures the demo app does not crash and is on the correct control page
     func testLaunch() throws {
         XCTAssertTrue(app.navigationBars.element(matching: NSPredicate(format: "identifier CONTAINS %@", controlName)).exists)
     }
+
+    func testBlocksPresentation() throws {
+        let blocksInteractionSwitch: XCUIElement = app.switches["Blocks interaction"]
+        let presentHUDButton: XCUIElement = app.buttons["Present HUD for 3 seconds"]
+        var blocksInteractionSwitchInitialValue = blocksInteractionSwitch.value
+
+        // switch value should start as 1
+        XCTAssert(blocksInteractionSwitchInitialValue as? String == "1")
+        presentHUDButton.tap()
+        // attempts to interact with "Blocks interaction" switch
+        blocksInteractionSwitch.tap()
+        // switch value should not change (remains 1)
+        XCTAssert(blocksInteractionSwitch.value as? String == blocksInteractionSwitchInitialValue as? String)
```

Comment:
Hmm does blocksInteractionSwitch.value == blocksInteractionSwitchInitialValue not work? Do we have to cast them both to strings? Or maybe blocksInteractionSwitch.value.equalTo(blocksInteractionSwitchInitialValue)?

Reply:
Unfortunately not because `blocksInteractionSwitch.value` is of type `Any?` so the only way to compare it is to cast it as a String.